src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterData.java

```diff
@@ -76,13 +85,9 @@ public TiffRasterData(final int width, final int height, final float[] data) {
      * @param value the value to be stored at the specified location;
      * potentially a Float&#46;NaN.
      */
-    public void setValue(final int x, final int y, final float value) {
-        if (x < 0 || x >= width || y < 0 || y >= height) {
-            throw new IllegalArgumentException(
-                "Coordinates out of range (" + x + ", " + y + ")");
-        }
-        data[y * width + x] = value;
-    }
+    public abstract void setValue(final int x, final int y, final float value);
```

Comment:
@gwlucastrig checkstyle is failing here saying these `final` parameters are redundant (See GitHub actions output).

If you'd like to execute the exact same target as GH Actions CI, take a look at this file in your local folder or GitHub UI: https://github.com/kinow/commons-imaging/blob/a87b545e1019b43bfbe31fa5df3a4f3095a8b438/.github/workflows/maven.yml#L41

To correct the problem,

1) check out your branch for this PR
2) fix the problem
3) push

The pull request will be updated, and the CI jobs will be re-triggered :+1:

Reply:
So I fixed the checkstyle complaints, but now the spotbugs code checker is producing an error.  I believe it is wrong, in this case. So I've been searching the web for a way to add suppressions to the code.  I imagine this is pretty common practice, and I will be updating the code as soon as I find a solution.

The problem is that two of my classes have arrays as member elements which they expose through a get method.  Spotbugs figures that this is an unsafe coding practice, and flags an error on the code...  The problem is that I WANT to do this "unsafe" coding practice.  I wish to avoid making redundant copies of potentially large data objects for reasons of both performance and, especially, memory conservation.  The whole idea is that the calling module obtains a TiffRasterDataFloat or TiffRasterDataInt object and can use it as a convenient container for the numberical raster arrays.

Reply:
Yup, there's a bunch of code already suppressed in the SpotBugs configuration. Just match the last number, whther EI_EXPOSE_REP**2** or EI_EXPOSE_REP**2** and it should pass.

https://github.com/apache/commons-imaging/blob/a87b545e1019b43bfbe31fa5df3a4f3095a8b438/src/conf/spotbugs-exclude-filter.xml#L158-L162

Reply:
Thanks.  I see it....  So much new stuff since the last time I worked on Apache code :-)